flink-table/flink-sql-client/src/test/resources/sql/table.q

```diff
@@ -1092,25 +1092,25 @@ Sink(table=[default_catalog.default_database.orders2], fields=[user, product, am
 | == Abstract Syntax Tree ==
 LogicalSink(table=[default_catalog.default_database.daily_orders], fields=[user, product, amount, dt])
-+- LogicalProject(user=[123:BIGINT], product=[_UTF-16LE'toothpick':VARCHAR(2147483647) CHARACTER SET "UTF-16LE"], amount=[1], dt=[_UTF-16LE'2022-06-12':VARCHAR(2147483647) CHARACTER SET "UTF-16LE"])
++- LogicalProject(user=[CAST(123:BIGINT):BIGINT], product=[CAST(_UTF-16LE'toothpick':VARCHAR(2147483647) CHARACTER SET "UTF-16LE"):VARCHAR(2147483647) CHARACTER SET "UTF-16LE"], amount=[CAST(1):INTEGER], dt=[CAST(_UTF-16LE'2022-06-12':VARCHAR(2147483647) CHARACTER SET "UTF-16LE"):VARCHAR(2147483647) CHARACTER SET "UTF-16LE"])
    +- LogicalValues(tuples=[[{ 0 }]])
 
 == Optimized Physical Plan ==
 Sink(table=[default_catalog.default_database.daily_orders], fields=[user, product, amount, dt])
-+- Calc(select=[123:BIGINT AS user, _UTF-16LE'toothpick':VARCHAR(2147483647) CHARACTER SET "UTF-16LE" AS product, 1 AS amount, _UTF-16LE'2022-06-12':VARCHAR(2147483647) CHARACTER SET "UTF-16LE" AS dt])
++- Calc(select=[CAST(123:BIGINT AS BIGINT) AS user, CAST(_UTF-16LE'toothpick':VARCHAR(2147483647) CHARACTER SET "UTF-16LE" AS VARCHAR(2147483647) CHARACTER SET "UTF-16LE") AS product, CAST(1 AS INTEGER) AS amount, CAST(_UTF-16LE'2022-06-12':VARCHAR(2147483647) CHARACTER SET "UTF-16LE" AS VARCHAR(2147483647) CHARACTER SET "UTF-16LE") AS dt])
    +- Values(type=[RecordType(INTEGER ZERO)], tuples=[[{ 0 }]])
 
 == Optimized Execution Plan ==
 Sink(table=[default_catalog.default_database.daily_orders], fields=[user, product, amount, dt])
-+- Calc(select=[123 AS user, 'toothpick' AS product, 1 AS amount, '2022-06-12' AS dt])
++- Calc(select=[CAST(123 AS BIGINT) AS user, CAST('toothpick' AS VARCHAR(2147483647)) AS product, CAST(1 AS INTEGER) AS amount, CAST('2022-06-12' AS VARCHAR(2147483647)) AS dt])
```

Comment:
Would be great to get a bit more explanation on the root cause of this change.

Reply:
this line is the reason https://github.com/apache/flink/pull/26827/files#diff-122600ca92f607b85c916ea9ed1c01e774a1556d328f68aef79b5df7a2787dfcR147

after that it takes nullability into account, and here is cast because of different nullability